### PR TITLE
Feature/eco 2300/crefopay credit card without 3d secure yves

### PR DESCRIPTION
**Pull request by nikita-lysenko**
Ticket: https://spryker.atlassian.net/browse/ECO-2504

### CODE REVIEW

File: src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
+declare var SecureFieldsClient: any;
+
+import Component from 'ShopUi/models/component';
+import ScriptLoader from 'ShopUi/components/molecules/script-loader/script-loader';
+
+const CREFOPAY_CONFIG = {
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.ts]: `CREFOPAY_CONFIG` -> `CREFO_PAY_CONFIG`.

File: src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    placeholders: {
+        accountHolder: "Your Name",
+        number: "0123456789101112",
+        cvv: "000"
+    }
+};
+
+export default class CrefopayFormLoader extends Component {
+    protected crefoPayScriptLoader: ScriptLoader;
+    protected secureFieldsClientInstance: any;
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.ts]: `secureFieldsClientInstance` -> `secureFieldsClient`.

File: src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                this.crefopayOrderId,
+                this.paymentRegisteredCallback,
+                this.initializationCompleteCallback,
+                CREFOPAY_CONFIG);
+        this.secureFieldsClientInstance.registerPayment();
+    }
+
+    protected paymentRegisteredCallback(response) {
+        if (response.resultCode === 0) {
+            // Successful registration, continue to next page using JavaScript
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.ts]: And why you do nothing on success?

File: src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        } else {
+            // Error during registration, check the response for more details and dynamically show a message for the customer
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.ts]: And who should implement error handling? Please show error message and error details.

File: src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        }
+    }
+
+    protected initializationCompleteCallback(response) {
+        if (response.resultCode === 0) {
+            // Successful registration, continue to next page using JavaScript
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.ts]: `Successful registration` - why registration? This is initializationComplete callback.

File: src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.ts [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        } else {
+            // Error during registration, check the response for more details and dynamically show a message for the customer
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.ts]: The same - please implement error handling.

File: src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.twig [hunk 7]
```diff
@@ -0,0 +1,26 @@
+{% extends model('component') %}
+
+{% define config = {
+    name: 'crefopay-form-loader',
+    tag: 'crefopay-form-loader'
+} %}
+
+{% define data = {
+    shopPublicKey: '',
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.twig]: Why do you need that data? If you can only leave only attribute.

File: src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.twig [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    orderId: ''
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.twig]: The same.

File: src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.twig (hunk 7 above)
Comment (tolerants) [reply src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.twig]: The only reason I see - to make it required. And as I see it should be required.

File: src/SprykerEco/Yves/CrefoPay/Theme/default/views/credit-card-3d/credit-card-3d.twig [hunk 9]
```diff
@@ -5,55 +5,29 @@
...
-            }
+        embed: {
+            form: data.form
         }
-    </script>
+    } only %}
+        {% block requestForm %}
+            <div>
+                {{ form_widget(embed.form.crefoPayCreditCard3D) }}
+                <label><input type="radio" data-crefopay="paymentMethod" value="CC">Creditcard</label><br>
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/views/credit-card-3d/credit-card-3d.twig]: Please remove `<br>`.

File: src/SprykerEco/Yves/CrefoPay/Theme/default/views/credit-card-3d/credit-card-3d.twig [hunk 10]
```diff
@@ -5,55 +5,29 @@
...
+                <span>Accountholder: </span>
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/views/credit-card-3d/credit-card-3d.twig]: Why you are mixing inline and block elements? Please fix markup.

File: src/SprykerEco/Yves/CrefoPay/Theme/default/views/credit-card-3d/credit-card-3d.twig (hunk 9 above)
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/views/credit-card-3d/credit-card-3d.twig]: Please add translation. Check it everywhere.

File: src/SprykerEco/Yves/CrefoPay/Theme/default/views/direct-debit/direct-debit.twig [hunk 11]
```diff
@@ -5,5 +5,24 @@
...
+    {% embed molecule('crefopay-form-loader', 'CrefoPay') with {
+        data: {
+            shopPublicKey: '',
+            orderId: ''
+        },
+        embed: {
+            form: data.form
+        }
+    } only %}
+    {% block requestForm %}
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/views/direct-debit/direct-debit.twig]: Wrong indention here and below.

**Review by tolerants — CHANGES_REQUESTED** [review]

File: src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.ts [hunk 12]
```diff
@@ -0,0 +1,100 @@
+declare var SecureFieldsClient: any;
+
+import Component from 'ShopUi/models/component';
+import ScriptLoader from 'ShopUi/components/molecules/script-loader/script-loader';
+
+export default class CrefopayFormLoader extends Component {
+    protected crefoPayScriptLoader: ScriptLoader;
+    public secureFieldsClient: any;
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.ts]: Why public? Also suggest to rename, bcs it's confusing we have vendor method and our class property with the same name.

File: src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.twig [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+    shopPublicKey: required,
+    orderId: required,
+    registerPaymentActionUrl: required,
+    secureFieldsLibraryUrl: required,
+    secureFieldsPlaceholders: {},
+    fields: []
+} %}
+
+{% if data.secureFieldsPlaceholders is not empty %}
+    {% set crefoPayConfig = {
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.twig]: It should be a part of data contract.

File: src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.twig [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+            number: data.secureFieldsPlaceholders.number,
+            cvv: data.secureFieldsPlaceholders.cvv
+        }
+    } %}
+{% endif %}
+
+{% define attributes = {
+    'shop-public-key': data.shopPublicKey,
+    'order-id': data.orderId,
+    'crefo-pay-config': data.secureFieldsPlaceholders ? crefoPayConfig | json_encode() : '',
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.twig]: But if we don't have `secureFieldsPlaceholders` then it looks like we don't need to run our js.

File: src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.twig [hunk 15]
```diff
@@ -0,0 +1,66 @@
+{% extends model('component') %}
+
+{% define config = {
+    name: 'crefopay-form-loader',
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.twig]: Please rename `crefopay-form-loader` -> `crefopay-payment-from`.

File: src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.twig [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+    } only %}
+    {% block requestForm %}
+        {{ form_widget(data.form.crefoPayCreditCard) }}
+
+        {% for field in data.fields %}
+            {{ _self.crefopayField(field.name, field.attribute, config.name) }}
+        {% endfor %}
+
+        <div class="{{ config.name }}__error {{ config.jsName }}__error spacing-y is-hidden">
+            {{ 'All fields are required.' | trans }}
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/components/molecules/crefopay-form-loader/crefopay-form-loader.twig]: Why not in glossary?

File: src/SprykerEco/Yves/CrefoPay/Theme/default/views/credit-card-3d/credit-card-3d.twig [hunk 17]
```diff
@@ -5,55 +5,19 @@
...
 
-        function paymentRegisteredCallback(response) {
-            if (response.resultCode === 0) {
-                // Successful registration, continue to next page using JavaScript
-            } else {
-                // Error during registration, check the response for more details and dynamically show a message for the customer
-            }
+    {% include molecule('crefopay-form-loader', 'CrefoPay') with {
+        data: {
+            form: data.form,
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/views/credit-card-3d/credit-card-3d.twig]: How is it working? In `crefopay-form-loader` we have 5 required fields, but you are using just 4 here.

File: src/SprykerEco/Yves/CrefoPay/Theme/default/views/direct-debit/direct-debit.twig [hunk 18]
```diff
@@ -5,5 +5,41 @@
...
+            context: _self
+        }
+    } only %}
+        {% block requestForm %}
+            {{ form_widget(data.form.crefoPayDirectDebit) }}
+            {% for field in embed.fields %}
+                {{ embed.context.crefopayField(field.name, field.attribute, field.placeholder) }}
+            {% endfor %}
+            <div class="spacing-y">
+                <label><input type="radio" data-crefopay="paymentInstrument.id" value="Aqc9O5SSJuQTDHC9KXFE0Q" class="spacing-right">Registered bank account (IBAN: DE12 3456 7890 1112 1314)</label>
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/views/direct-debit/direct-debit.twig]: Why not translated?

**Review by tolerants — CHANGES_REQUESTED** [review]

File: src/SprykerEco/Yves/CrefoPay/Theme/default/views/credit-card-3d/credit-card-3d.twig (hunk 17 above)
Comment (nikita-lysenko) [reply src/SprykerEco/Yves/CrefoPay/Theme/default/views/credit-card-3d/credit-card-3d.twig]: it's just placeholders for further implementation, fixed for current state

File: src/SprykerEco/Yves/CrefoPay/Theme/default/views/direct-debit/direct-debit.twig (hunk 18 above)
Comment (nikita-lysenko) [reply src/SprykerEco/Yves/CrefoPay/Theme/default/views/direct-debit/direct-debit.twig]: added translation in macro at line 10-11

File: src/SprykerEco/Yves/CrefoPay/Theme/default/components/atoms/crefopay-checkbox-helper/crefopay-checkbox-helper.ts [hunk 19]
```diff
@@ -0,0 +1,67 @@
...
+        this.mapEvents();
+    }
+
+    protected mapEvents(): void {
+        this.triggers.forEach((trigger: HTMLElement) => {
+            trigger.addEventListener('change', (event: Event) => this.onChange(event));
+        });
+    }
+
+    protected applyAttributes (): void {
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/components/atoms/crefopay-checkbox-helper/crefopay-checkbox-helper.ts]: Please remove space before `()`.

File: src/SprykerEco/Yves/CrefoPay/Theme/default/components/atoms/crefopay-checkbox-helper/crefopay-checkbox-helper.ts [hunk 20]
```diff
@@ -0,0 +1,67 @@
...
+
+    protected onChange(event: Event): void {
+        this.targets.forEach((target: HTMLInputElement)=>{
+            target.checked = false;
+        });
+        this.syncCheckboxes(event.currentTarget);
+    }
+
+    protected syncCheckboxes(triggered: EventTarget): void {
+        let jointContainer = (<HTMLElement>triggered).closest(this.jointContainerSelector);
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/components/atoms/crefopay-checkbox-helper/crefopay-checkbox-helper.ts]: `let` -> `const`.

File: src/SprykerEco/Yves/CrefoPay/Theme/default/components/atoms/crefopay-checkbox-helper/crefopay-checkbox-helper.ts [hunk 21]
```diff
@@ -0,0 +1,67 @@
...
+        let triggeredTarget: HTMLInputElement = jointContainer.querySelector(this.targetSelector);
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/components/atoms/crefopay-checkbox-helper/crefopay-checkbox-helper.ts]: `let` -> `const`.

File: src/SprykerEco/Yves/CrefoPay/Theme/default/components/atoms/crefopay-checkbox-helper/crefopay-checkbox-helper.ts [hunk 22]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    protected mapEvents(): void {
+        this.triggers.forEach((trigger: HTMLElement) => {
+            trigger.addEventListener('change', (event: Event) => this.onChange(event));
+        });
+    }
+
+    protected applyAttributes (): void {
+        this.targets.forEach((target: HTMLInputElement)=>{
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/components/atoms/crefopay-checkbox-helper/crefopay-checkbox-helper.ts]: Wrap `=>` with spaces.

File: src/SprykerEco/Yves/CrefoPay/Theme/default/components/atoms/crefopay-checkbox-helper/crefopay-checkbox-helper.ts [hunk 23]
```diff
@@ -0,0 +1,67 @@
...
+            target.setAttribute(this.customAttrName, this.customAttrValue);
+        })
+    }
+
+    protected onChange(event: Event): void {
+        this.targets.forEach((target: HTMLInputElement)=>{
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/components/atoms/crefopay-checkbox-helper/crefopay-checkbox-helper.ts]: Wrap `=>` with spaces.

File: src/SprykerEco/Yves/CrefoPay/Theme/default/components/atoms/crefopay-checkbox-helper/crefopay-checkbox-helper.ts [hunk 24]
```diff
@@ -0,0 +1,67 @@
...
+
+    get containerSelector() {
+        return this.getAttribute('payment-container-selector');
+    }
+
+    get targetSelector() {
+        return this.getAttribute('target-selector');
+    }
+
+    get customAttrName() {
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/components/atoms/crefopay-checkbox-helper/crefopay-checkbox-helper.ts]: `customAttrName` -> `customAttributeName`.

File: src/SprykerEco/Yves/CrefoPay/Theme/default/components/atoms/crefopay-checkbox-helper/crefopay-checkbox-helper.ts [hunk 25]
```diff
@@ -0,0 +1,67 @@
...
+        return this.getAttribute('custom-attr-name');
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/components/atoms/crefopay-checkbox-helper/crefopay-checkbox-helper.ts]: `custom-attr-name` -> `custom-attribute-name`.

File: src/SprykerEco/Yves/CrefoPay/Theme/default/components/atoms/crefopay-checkbox-helper/crefopay-checkbox-helper.ts [hunk 26]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    get customAttrValue() {
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/components/atoms/crefopay-checkbox-helper/crefopay-checkbox-helper.ts]: `customAttrValue` -> `customAttributeValue`.

File: src/SprykerEco/Yves/CrefoPay/Theme/default/components/atoms/crefopay-checkbox-helper/crefopay-checkbox-helper.ts [hunk 27]
```diff
@@ -0,0 +1,67 @@
...
+        return this.getAttribute('custom-attr-value');
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/components/atoms/crefopay-checkbox-helper/crefopay-checkbox-helper.ts]: `custom-attr-value` -> `custom-attribute-value`.

File: src/SprykerEco/Yves/CrefoPay/Theme/default/components/atoms/crefopay-checkbox-helper/crefopay-checkbox-helper.ts [hunk 28]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    protected onChange(event: Event): void {
+        this.targets.forEach((target: HTMLInputElement)=>{
+            target.checked = false;
+        });
+        this.syncCheckboxes(event.currentTarget);
+    }
+
+    protected syncCheckboxes(triggered: EventTarget): void {
```
Comment (tolerants) [inline src/SprykerEco/Yves/CrefoPay/Theme/default/components/atoms/crefopay-checkbox-helper/crefopay-checkbox-helper.ts]: Why the name is `syncCheckboxes`? It is working only with one checkbox. And it not syncs but makes it checked. Suggest to rename into `checkCheckbox`.
`triggered`-> `checkbox`.

**Review by tolerants — CHANGES_REQUESTED** [review]

**Review by tolerants — APPROVED** [review]

**Review by lockw1n — APPROVED** [review]
Changes were tested.

**Pull request merged**